server/routes/stripe.js

```diff
@@ -0,0 +1,117 @@
+const router = require("express").Router();
+const User = require("../schemas/users");
+const Order = require("../schemas/orders");
+const Address = require("../schemas/address");
+const stripe = require("stripe")(process.env.STRIPE_KEY);
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+const logger = require("../utils/logger");
+
+router.post("/charge", async (req, res, next) => {
+  let amount = parseFloat(req.body.total) + (req.body.details.shippingCost ? parseFloat(req.body.details.shippingCost) : 0);
+
+  const chargedCustomer = await stripe.charges.create({
+    amount: amount * 100,
+    currency: "CAD",
+    source: req.body.token
+  });
+
+  if (!chargedCustomer) {
+    return res.status(400).send({ error: error });
+  }
+
+  
+  const userId = req.body.user;
+  let user;
+
+  logger.log("info", `User id is ${userId}`);
+  if (userId) {
+    user = await User.findOne(
+      { _id: userId },
+      { __v: false, password: false, date: false }
+    );
+    if (!user) return res.status(400).send({ error: "User is not found..." });
+    logger.log("info", "The user details is", user.toJSON());
+  } else {
+    user = await User.findOne(
+      { email: req.body.details.email },
+      { __v: false, password: false, date: false }
+    );
+    if (!user) {
+      const salt = await bcrypt.genSalt(10);
```

Comment:
I think this is an interesting idea... however I feel like we shouldn't save user details unless they consent, itd be more like we should say do you want to create an account and then save the address and user and stuff.. if they dont, then we just save the order with no user

Reply:
Removed the user creation